osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/application/config/OslpConfig.java

```diff
@@ -0,0 +1,245 @@
+import org.opensmartgridplatform.shared.infra.networking.DisposableNioEventLoopGroup;
+import org.opensmartgridplatform.shared.security.CertificateHelper;
+import org.opensmartgridplatform.webdevicesimulator.service.OslpChannelHandler;
+import org.opensmartgridplatform.webdevicesimulator.service.OslpSecurityHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+@Configuration
+public class OslpConfig {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OslpConfig.class);
+
+    private static final String PROPERTY_NAME_OSLP_TIMEOUT_CONNECT = "oslp.timeout.connect";
+
+    private static final String PROPERTY_NAME_OSLP_PORT_CLIENT = "oslp.port.client";
+    private static final String PROPERTY_NAME_OSLP_ELSTER_PORT_CLIENT = "oslp.elster.port.client";
+    private static final String PROPERTY_NAME_OSLP_PORT_SERVER = "oslp.port.server";
+    private static final String PROPERTY_NAME_OSLP_ELSTER_PORT_SERVER = "oslp.elster.port.server";
+    private static final String PROPERTY_NAME_OSLP_ADDRESS_CLIENT = "oslp.address.client";
+
+    private static final String PROPERTY_NAME_OSLP_SECURITY_SIGNKEY_PATH = "oslp.security.signkey.path";
+    private static final String PROPERTY_NAME_OSLP_SECURITY_VERIFYKEY_PATH = "oslp.security.verifykey.path";
+    private static final String PROPERTY_NAME_OSLP_SECURITY_KEYTYPE = "oslp.security.keytype";
+    private static final String PROPERTY_NAME_OSLP_SECURITY_SIGNATURE = "oslp.security.signature";
+    private static final String PROPERTY_NAME_OSLP_SECURITY_PROVIDER = "oslp.security.provider";
+    private static final String PROPERTY_NAME_OSLP_SEQUENCE_NUMBER_WINDOW = "oslp.sequence.number.window";
+    private static final String PROPERTY_NAME_OSLP_SEQUENCE_NUMBER_MAXIMUM = "oslp.sequence.number.maximum";
+
+    @Resource
+    private Environment environment;
+
+    @Bean(name = "deviceSimulatorOslpNettyServerBossGroup")
+    public DisposableNioEventLoopGroup serverBossGroup() {
+        // TODO - Should we pass in any parameters, like for example number of
+        // threads, to the constructor?
+        return new DisposableNioEventLoopGroup();
+    }
+
+    @Bean(name = "deviceSimulatorOslpNettyServerWorkerGroup")
+    public DisposableNioEventLoopGroup serverWorkerGroup() {
+        // TODO - Should we pass in any parameters, like for example number of
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)